hawk/src/main/java/com/orhanobut/hawk/Hawk.java

```diff
@@ -111,20 +102,90 @@ public static void init(Context context, String password, LogLevel logLevel) {
      * @return true if put is successful
      */
     public static <T> boolean put(String key, List<T> list) {
+        return put(key, null, list, true, true) != null;
+    }
+
+    /*
+     * A helper put method.
+     *
+     * @param key          the key
+     * @param value        the value if isList is false, otherwise null
+     * @param list         the list, if isList is true, otherwise null
+     * @param isList       determines if a list instance should be present
+     * @param addToStorage determines if the keyval should be immediately added to storage
+     */
+    private static <T> String put(String key, T value, List<T> list, boolean isList,
+                                  boolean addToStorage) {
+        ensureKeyIsValid(key);
+        if (isList) {
+            ensureListIsValid(list);
+        } else {
+            ensureValueIsValid(value);
+        }
+
+        String cipherText;
+        if (isList) {
+            cipherText = encoder.encode(list);
+        } else {
+            cipherText = encoder.encode(value);
+        }
+
+        if (cipherText == null) {
```

Comment:
Indeed, we should not accept null values at all. I'll add another comment for the atomicity as well.